pyecsca/sca/attack/DPA.py

```diff
@@ -0,0 +1,77 @@
+from pyecsca.ec.mult import ScalarMultiplier
+from pyecsca.sca.trace import Trace
+from pyecsca.ec.point import Point
+from pyecsca.ec.context import DefaultContext, local
+from pyecsca.ec.params import DomainParameters
+from pyecsca.sca.trace import Trace, average, subtract
+from pyecsca.sca.trace.plot import plot_trace
+from typing import Tuple
+import holoviews as hv
+
+class DPA():
+
+    traces: list[Trace]
+    points: list[Point]
+    mult: ScalarMultiplier
+    params: DomainParameters
+
+    def __init__(self, points: list[Point], traces: list[Trace], mult: ScalarMultiplier, params: DomainParameters):
+        self.points = points
+        self.traces = traces
+        self.mult = mult
+        self.params = params
+
+    def compute_split_point(self, guessed_scalar: int, target_bit: int, point: Point) -> Point:
+        with(local(DefaultContext())) as ctx:
+            self.mult.init(self.params, point)
+            self.mult.multiply(guessed_scalar)
+        action_index = -1
+        for bit in bin(guessed_scalar)[2:target_bit + 2]:
+            if bit == '1':
+                action_index += 2
+            elif bit == '0':
+                action_index += 1
+        result = ctx.actions.get_by_index([0, action_index])[0]
+        return result.output_points[0]
+
+    def split_traces(self, guessed_scalar: int, target_bit: int) -> Tuple[list[Trace], list[Trace]]:
+        one_traces = []
+        zero_traces = []
+        for i in range(len(self.points)):
+            split_value = self.compute_split_point(guessed_scalar, target_bit, self.points[i]).coords.get('X')
```

Comment:
```suggestion
            split_value = self.compute_split_point(guessed_scalar, target_bit, self.points[i]).X
```
But this might not work for some coordinate models that do not have an "X" coordinate. Just for future reference there should be a TODO comment explaining this.